src/rpdk/core/contract/resource_client.py

```diff
@@ -268,17 +268,17 @@ def convert_type(document_input, transformed_property):
     def compare(document_input, document_output):
         try:
             if isinstance(document_input, str):
-                return re.match(f"^{document_input}$", document_output)
+                return bool(re.match(f"^{document_input}$", document_output))
```

Comment:
could u provide and example of what you are trying to achieve? 
would like to see what form `document_input` takes and what is `document_output`. 

I can see that it is used as:
```python
resource_client.compare(
    pruned_input_model[key],
    pruned_output_model[key]
)
```

is **key** a property name, if so then why do you need to use regex match

Reply:
Lets say a customer adds the name of a property when creating the resource. This property name is updated to an ARN. ```propertyTransform ``` property in the schema will allow one to define the arn regex to ensure this property is not falsely drifted.

```
document_input: {
  "propertyName": "Name"
}
```

```
document_output: {
  "propertyName": "arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:ecs:[a-z]{2}[-]{1}[a-z]{4,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}cluster/Name"
}
```

where the propertyTransform in the schema is:
```
"propertyTransform": {
    "/properties/propertyName": "$join([\"arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:ecs:[a-z]{2}[-]{1}[a-z]{4,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}cluster\\/\", PropertyName])"
  }
```

Reply:
great details, would be great to add it to PR details. 

but still trying to understand where regex is coming from? from property pattern like:
```json
{
  "properties": {
    "propertyA": {
      "type": "...",
      "pattern": "^arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:ecs:[a-z]{2}[-]{1}[a-z]{4,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}cluster/Name$"
    }
  }
}
```
or it is coming from transform and is used only on comparison during contract test execution?

Reply:
It is coming from propertyTransform.

```
"propertyTransform": {
    "/properties/propertyName": "$join([\"arn:(aws)[-]{0,1}[a-z]{0,2}[-]{0,1}[a-z]{0,3}:ecs:[a-z]{2}[-]{1}[a-z]{4,10}[-]{1}[1-3]{1}:[0-9]{12}[:]{1}cluster\\/\", PropertyName])"
  }
```

Adding the pattern check will not allow customers to set the string ```Name``` for the property. ```propertyTransform``` is primarily used by Drift. Contract test is just replicating the drift functionality for testing purposes.